### PR TITLE
Update: Add support for the next controller when the session state is not found

### PR DESCRIPTION
**Pull request by akku-aakash**
#### Description

- Terminates `authCallback` if the state is not found.
- Return the state during session verification.
- Handles the authentication redirect after verifying the session.

### CODE REVIEW

File: oidc-consumer/src/index.ts [hunk 1]
```diff
@@ -194,11 +194,14 @@ class OidcConsumer {
   async authCallback(request: Request, response: Response, next: NextFunction, queryParams: Object, httpOptions?: WreckHttpOptions) {
     const { code, state } = request.query;
 
-    const sessionState = (request.session as ICustomSession).state;
-    if (!sessionState) {
-      console.log("Verifying session...")
-      await this.verifySession(request, response, next);
+    if (!(request.session as ICustomSession).state) {
+      console.log("Reloading session because no state in session.");
+      await this.loadSession(request, response, next);
```
Comment (rickygarg) [inline oidc-consumer/src/index.ts]: instead, just send session, and next can be handled by this method

File: oidc-consumer/test/index.test.ts [hunk 2]
```diff
@@ -33,34 +33,70 @@ const consumer = new OidcConsumer({
...
-    it('should throw an error if session state is not present and throwError is true', async () => {
+    it('should throw an error if session.reload fails (rejects)', async () => {
       const req = mockReq();
-      const res = mockRes();
-      req.session = {reload: jest.fn()};
+      req.session = { reload: jest.fn((cb) => cb(new Error('Failed to load session')))};
+  
+      try {
+        await consumer.loadSession(req.session, true);
+        expect(true).toBe(true);
```
Comment (paras-fundwave) [inline oidc-consumer/test/index.test.ts]: please remove

File: oidc-consumer/test/index.test.ts [hunk 3]
```diff
@@ -33,34 +33,70 @@ const consumer = new OidcConsumer({
...
+        await consumer.loadSession(req.session, false);
+        expect(true).toBe(true);
+      } catch (error) {
+        expect(error).toEqual('SESSION_LOAD_FAILED');
+      }
+  
+      expect(req.session.reload).toHaveBeenCalledTimes(1);
+    });
+  
+    it('should retry once when session state is missing and retryOnFailure is true', async () => {
```
Comment (paras-fundwave) [inline oidc-consumer/test/index.test.ts]: this is covered in the case above
can remove

File: oidc-consumer/test/index.test.ts [hunk 4]
```diff
@@ -33,34 +33,70 @@ const consumer = new OidcConsumer({
...
 
-        await consumer.verifySession(req, res, next, true);
-        expect(next.calledOnce).toBe(true);
-        expect(next.firstCall.args[0]).toBeInstanceOf(Error);
-        expect(next.firstCall.args[0].message).toEqual('SESSION_VERIFICATION_FAILED');
+      await consumer.loadSession(req.session, true);
+      expect(req.session.reload).toHaveBeenCalledTimes(2);
     });
+  
+    it('should throw an error after retry if session state is still missing and retryOnFailure is true', async () => {
```
Comment (paras-fundwave) [inline oidc-consumer/test/index.test.ts]: can remove as this has been covered in previous cases

**Review by paras-fundwave — COMMENTED** [review]
looks good, needs minor changes

**Review by paras-fundwave — DISMISSED** [review]
lgtm

nit: should the error message be in sentence-case?

**Review by rickygarg — APPROVED** [review]

**Pull request merged**